src/main/java/org/spongepowered/common/event/tracking/CauseTracker.java

```diff
@@ -623,6 +623,7 @@ public boolean spawnEntity(World world, Entity entity) {
             // capture all entities until the phase is marked for completion.
             if (!isForced) {
                 try {
+                    Sponge.getCauseStackManager().pushCause(world);
```

Comment:
Why are we pushing a world here? It's not always the world's cause to be spawning an entity, it should be up to the phase state to push the world cause.